lib/matplotlib/axes/_axes.py

```diff
@@ -3281,27 +3281,12 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         kwargs = {k: v for k, v in kwargs.items() if v is not None}
         kwargs.setdefault('zorder', 2)
 
-        self._process_unit_info([("x", x), ("y", y)], kwargs, convert=False)
-
-        # Make sure all the args are iterable; use lists not arrays to preserve
-        # units.
-        if not np.iterable(x):
-            x = [x]
-
-        if not np.iterable(y):
-            y = [y]
-
+        x, y, xerr, yerr = self._process_unit_info(
+            [("x", x), ("y", y), ("x", xerr), ("y", yerr)], kwargs)
+        x, y = np.atleast_1d(x, y)  # Make sure all the args are iterable.
```

Comment:
Do you not need to do this for `x/yerr` as well?  I don't see where you handle the `xerr` is just a single number case, though maybe it just works via numpy arithmetic?

Reply:
That's handled by numpy being able to do `array + number`.